libs/execution_context/ebpf_maps.c

```diff
@@ -53,23 +53,40 @@ typedef struct _ebpf_core_object_map
  * cold list is trimmed by removing the oldest entries in the list, which are part of the oldest generation. The benefit
  * is that the lock only needs to be acquired when moving items between generations, and not on every access.
  */
-typedef struct _ebpf_lru_entry
+typedef struct _ebpf_lru_entry_partition
 {
     ebpf_list_entry_t list_entry; //< List entry for the hot or cold list.
     size_t generation;            //< Generation in which the key was last accessed.
-    uint8_t key[1];               //< Variable length key. The actual size is determined by the map definition.
+    size_t padding;               //< Make the structure size a multiple of 16 bytes.
```

Comment:
should we add a compile-time assert for this?

Reply:
also, depending on how many different compilers you need to target, you should be able to use a compiler keyword to create padding to a specified alignment automatically.